make.paws/tests/testthat/test_handlers_restxml.R

```diff
@@ -0,0 +1,692 @@
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input6(
+    PipelineId = "id",
+    QueryDoc = list(
+      fizz = c("buzz", "pop"),
+      foo = c("bar", "baz")
+    )
+  )
+  req <- new_request(svc, op6, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/2014-01-01/jobsByPipeline/id?fizz=buzz&fizz=pop&foo=bar&foo=baz")
+})
+
+test_that("query string with bool (true)", {
+  op7 <- Operation(
+    name = "OperationName",
+    http_method = "GET",
+    http_path = "/path"
+  )
+  op_input7 <- function(BoolQuery) {
+    args <- list(BoolQuery = BoolQuery)
+    interface <- Structure(
+      BoolQuery = Scalar(type = "boolean", .attrs = list(location = "querystring", locationName = "bool-query"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input7(
+    BoolQuery = TRUE
+  )
+  req <- new_request(svc, op7, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/path?bool-query=true")
+})
+
+test_that("query string with bool (false)", {
+  op8 <- Operation(
+    name = "OperationName",
+    http_method = "GET",
+    http_path = "/path"
+  )
+  op_input8 <- function(BoolQuery) {
+    args <- list(BoolQuery = BoolQuery)
+    interface <- Structure(
+      BoolQuery = Scalar(type = "boolean", .attrs = list(location = "querystring", locationName = "bool-query"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input8(
+    BoolQuery = FALSE
+  )
+  req <- new_request(svc, op8, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/path?bool-query=false")
+})
+
+test_that("URI and query string parameters", {
+  op9 <- Operation(
+    name = "OperationName",
+    http_method = "GET",
+    http_path = "/2014-01-01/jobsByPipeline/{PipelineId}"
+  )
+  op_input9 <- function(Ascending, PageToken, PipelineId) {
+    args <- list(Ascending = Ascending, PageToken = PageToken, PipelineId = PipelineId)
+    interface <- Structure(
+      Ascending = Scalar(type = "string", .attrs = list(location = "querystring", locationName = "Ascending")),
+      PageToken = Scalar(type = "string", .attrs = list(location = "querystring", locationName = "PageToken")),
+      PipelineId = Scalar(type = "string", .attrs = list(location = "uri", locationName = "PipelineId"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input9(
+    Ascending = "true",
+    PageToken = "bar",
+    PipelineId = "foo"
+  )
+  req <- new_request(svc, op9, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/2014-01-01/jobsByPipeline/foo?Ascending=true&PageToken=bar")
+})
+
+
+#-------------------------------------------------------------------------------
+
+# XML tests
+
+test_that("Basic XML Case1", {
+  op_test <- Operation(name = "OperationName")
+  op_input_test <- function(OperationRequest) {
```

Comment:
Hmmmm. Looking at line 77 [https://github.com/aws/aws-sdk-go/blob/master/private/protocol/xml/xmlutil/build.go](url)
I'm wondering if the more suitable interface would be:
```
interface <- Structure(
      _ = Structure(
          Description = Scalar(type = "string"),
          Name = Scalar(type = "string"),
          .attrs = list(locationName = "OperationRequest", xmlURI = "https://foo/")
     )
)
```
Which then gets back to the underscore issue

Reply:
That is feasible but at least right now, the tags like locationName that would be attached to `shape["_"]` are currently attached to `shape`. Also, that would change the calling signature of the operation -- it would require your call to the operation be `operation(_ = list(Description = "foo", Name = "bar"))`.